### PR TITLE
Update Python Publish workflow

### PR DESCRIPTION
**Pull request by dpgraham4401**
## Description
This PR adds an updated python_publish. Tested that it does work on test PyPi. add pyproject.toml, remove setup.cfg (was hoping could just have 1 file but still ended up having to add requirements.txt, bummer)

@wiljnichepa 
This only thing I haven't tested well is the path filter on the event triggers. It should only run when on a PR where 'pyproject.toml' is changed but I guess we'll figure out next time someone makes a change.

### CODE REVIEW

File: emanifest-py/pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+issues = "https://github.com/USEPA/e-Manifest/issues"
+documentation = "https://github.com/USEPA/e-Manifest/emanifest-py"
+homepage = "https://github.com/USEPA/e-Manifest"
+repository = "https://github.com/USEPA/e-Manifest"
+
 [build-system]
 requires = [
     "setuptools>=42",
-    "wheel"
+    "wheel",
```
Comment (wiljnichepa) [inline emanifest-py/pyproject.toml:34]: Does this trailing comma (line 34, after "wheel") cause any problems?

File: emanifest-py/README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@
   - [Methods](#methods)
   - [Help](#help)
 - [Contact](#contact)
-- [License](LICENSE.txt)
+- [License](LICENSE)
```
Comment (wiljnichepa) [inline emanifest-py/README.md:20]: I tested this without the .txt extension and didn't get it to link correctly to the license page. I think we should keep the extension - please let me know if I messed up and it's fine

File: emanifest-py/pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,36 @@
+[project]
+name = "emanifest"
+version = "2.0.7"
```
Comment (wiljnichepa) [inline emanifest-py/pyproject.toml:3]: In the next fix, I'd like to see if we can auto-increment the package number. I'm going to create a new issue for actions work like this.

**Review by wiljnichepa — COMMENTED** [review]
This looks great; thanks for getting to this so quickly. My only concerns are about the trailing comma in the setuptools requirements in the pyproject.toml and the license.txt extension in the readme.

File: emanifest-py/pyproject.toml (hunk 1 above)
Comment (dpgraham4401) [reply emanifest-py/pyproject.toml:34]: [trailing commas are A-OK in toml arrays](https://github.com/toml-lang/toml/issues/246#issuecomment-57919189) but I know they can bug people so if you want to adjust, I'm A-OK with that. I didn't have any problems building with the current condifs

Comment (wiljnichepa) [reply emanifest-py/pyproject.toml:34]: Doesn't bother me - if it works, no need to mess with it

File: emanifest-py/README.md (hunk 2 above)
Comment (dpgraham4401) [reply emanifest-py/README.md:20]: TLDR we can change it back if you want, just let me know. 

hmmm that's weird. I can't imagine the file extension is the problem. 

The only reason I changed it  was b/c I was debugging an error about "LICENSE.txt" being an unrecognized format or something but it turns out that file wasn't the problem at all.

File: emanifest-py/pyproject.toml (hunk 3 above)
Comment (dpgraham4401) [reply emanifest-py/pyproject.toml:3]: Very cool, curious to see how we can do this.

File: emanifest-py/README.md (hunk 2 above)
Comment (wiljnichepa) [reply emanifest-py/README.md:20]: Yeah, let's go ahead and change it back. If it catches fire in the next release we know where to start with the fix.

Comment (dpgraham4401) [reply emanifest-py/README.md:20]: @wiljnichepa 
If it sets you mind at ease, you can test that the README links to the LICENSE file on this branch here
https://github.com/dpgraham4401/e-manifest/tree/publish_workflow/emanifest-py

Doesn't appear to have any problems. 

If I don't hear back from you, i'll change it to LICENSE.txt

**Comment by dpgraham4401** [comment]
Thanks for the input @wiljnichepa I went ahead and squashed and merged after making your recommended changes.

**Pull request merged**